app/src/main/java/kth/jjve/xfran/EditProfileActivity.java

```diff
@@ -0,0 +1,150 @@
+            super.onBackPressed();
+        }
+    }
+
+    private void initNavMenu() {
+        profileMenu = navigationView.getMenu();
+        profileMenu.findItem(R.id.nav_login).setVisible(false);
+        profileMenu.findItem(R.id.nav_logout).setVisible(true);
+        profileMenu.findItem(R.id.nav_profile).setVisible(true);
+
+        navigationView.bringToFront();
+        ActionBarDrawerToggle toggle = new ActionBarDrawerToggle(this, drawerLayout, toolbar,
+                R.string.nav_open, R.string.nav_close);
+        drawerLayout.addDrawerListener(toggle);
+        toggle.syncState();
+        navigationView.setNavigationItemSelectedListener(this);
+        navigationView.setCheckedItem(R.id.nav_home);
+    }
+
+    @Override
+    public boolean onNavigationItemSelected(@NonNull MenuItem item) {
+        int id = item.getItemId();
+        if (id == R.id.nav_home) {
+            startActivity(new Intent(EditProfileActivity.this, HomeScreenActivity.class));
+            finish();
+        } else if (id == R.id.nav_login) {
+            startActivity(new Intent(EditProfileActivity.this, LoginActivity.class));
+        } else if (id == R.id.nav_logout) {
+            //Todo: make sure that logout works everywhere
+            Toast.makeText(EditProfileActivity.this, "Log out only in homescreen", Toast.LENGTH_SHORT).show();
+            startActivity(new Intent(EditProfileActivity.this, HomeScreenActivity.class));
+            finish();
+        }
+        drawerLayout.closeDrawer(GravityCompat.START);
+        return true;
+    }
+
+    @SuppressLint("SetTextI18n")
+    private void setViews(UserProfile userProfile){
+        if (!userProfile.checkEmpty()){
+            mFirstName.setText(userProfile.getFirstName());
+            mLastName.setText(userProfile.getLastName());
+            mEmail.setText(userProfile.getEmail());
+            mWeight.setText(Double.toString(userProfile.getWeight()));
+            mLength.setText(Integer.toString(userProfile.getHeight()));
+        }
+    }
+
+    private void saveUserProfile(){
+        try {
+            // Since a double/int can't be directly read from a edittext,
+            // it is first read as string and then parsed
+            String w = mWeight.getText().toString();
+            String l = mLength.getText().toString();
+            double weight = Double.parseDouble(w);
+            int length = Integer.parseInt(l);
+
+            mUserProfileVM.setUserProfile(
+                    mFirstName.getText().toString(),
+                    mLastName.getText().toString(),
+                    mEmail.getText().toString(),
+                    weight,
+                    length);
+
+            Toast.makeText(getApplicationContext(), "User profile saved", Toast.LENGTH_SHORT).show();
+            finish();
+        } catch (Exception e) {
+            e.printStackTrace();
+            Toast.makeText(getApplicationContext(), "Edittexts cannot be empty", Toast.LENGTH_SHORT).show();
```

Comment:
I think this block has a bug. When I delete my first and last name it doesn't throw an exception and weird things happen - it saves, but it doesn't change on profile activity, but it still appears empty if I return to edit profile activity.